### PR TITLE
implement put method for traveler-groups (controller,service, tests)

### PR DESCRIPTION
**Pull request by ZeshanShahid79**

### CODE REVIEW

File: backend/src/main/java/capstoneexpensetracker/backend/travelergroup/TravelerGroupService.java [hunk 1]
```diff
@@ -38,4 +39,12 @@ public void deleteTravelerGroup(String id) {
         }
         travelerGroupRepository.deleteById(id);
     }
+
+    public TravelerGroup updateTravelerGroupById(String id, TravelerGroup travelerGroup) {
+        Optional<TravelerGroup> travelerGroupToFind = travelerGroupRepository.findById(id);
+        if (travelerGroupToFind.isPresent()) {
+            return travelerGroupRepository.save(travelerGroup);
```
Comment (ChristianBorn) [inline backend/src/main/java/capstoneexpensetracker/backend/travelergroup/TravelerGroupService.java:46]: Your repository has an implicit method existsById() which you could also use here

File: frontend/src/HomePage.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import AddTravelerGroupForm from "./components/TravelerGroup/AddTravelerGroupForm/AddTravelerGroupForm";
+import AddTravelerForm from "./components/Traveler/AddTravelerForm/AddTravelerForm";
+import TravelerOverview from "./components/Traveler/TravelerOverview";
+import "./Homepage.css";
+
+export default function HomePage() {
+  const [travelerGroupList, setTravelerGroupList] = useState<
+    TravelerGroupModel[]
+  >([]);
+  const [travelers, setTravelers] = useState<TravelerModel[]>([]);
```
Comment (ChristianBorn) [inline frontend/src/HomePage.tsx:16]: I think you can omit the [] as your initial value of the useState

File: frontend/src/HomePage.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        return response.data;
+      })
+      .then(setTravelerGroupList);
+  };
+  const fetchAllTravelers = () => {
+    axios
+      .get("/api/travelers")
+      .then((response) => response.data)
+      .then(setTravelers);
+  };
```
Comment (ChristianBorn) [inline frontend/src/HomePage.tsx:36]: Consider having catch-statements in your axios call like
`catch((error) => console.error("Error while getting xyz:" + error))`

File: frontend/src/Homepage.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    width: 100vw;
+    overflow: scroll;
+    padding-left: 12px;
+    padding-right: 12px;
+    border: black solid 1px;
+
+
+}
+
+.NavLink {
```
Comment (ChristianBorn) [inline frontend/src/Homepage.css:18]: I'm not sure if the NavLinks behave like links if you set a custom class. If not, consider adding `cursor: pointer`

**Review by ChristianBorn — APPROVED** [review]

**Review by Dskottke — APPROVED** [review]

**Pull request merged**